File: abc065/src/c.cpp
#include <stdio.h>
#define MOD 1000000007
#define MAX(x, y) ((x) > (y) ? (x) : (y))
#define MIN(x, y) ((x) < (y) ? (x) : (y))
#define ABS(x) ((x) < 0 ? -(x) : (x)) // absolute value
typedef long long ll;
ll _fact(ll r)
{
    ll ans = 1;
    for (ll i = 1; i <= r; i++) {
        ans = ans * i % MOD;
    }
    return ans;
}
int main()
{
    int n, m;
    scanf("%d%d", &n, &m);
    if (ABS(n - m) >= 2) {
        printf("0\n");
        return 0;
    } // [1]

    ll ans = 0;
    if (n == m) {
        ans = (_fact(n) * _fact(m) % MOD) * 2 % MOD;
    } // [3]
    else {
        ans = (_fact(n) * _fact(m) % MOD);
    } // [2]
    printf("%lld\n", ans);
    return 0;
}

File: abc065/src/c.py
n, m = map(int, input().split())
MOD = 1000000007 // modulo


def nPk(n, k):
    r = 1
    while k > 0:
        r *= n
        r %= MOD
        n -= 1
        k -= 1
    return r


if abs(n - m) > 1:
    print(0)
elif n == m:
    print(nPk(n, n) * nPk(m, m) * 2 % MOD)
else:
    print(nPk(n, n) * nPk(m, m) % MOD)
